journal.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+from cryptacular.bcrypt import BCRYPTPasswordManager
+from sqlalchemy.orm import scoped_session, sessionmaker
+from zope.sqlalchemy import ZopeTransactionExtension
+import os
+from pyramid.config import Configurator
+from pyramid.view import view_config
+from pyramid.security import remember, forget
+from waitress import serve
+import sqlalchemy as sa
+from sqlalchemy.ext.declarative import declarative_base
+from pyramid.httpexceptions import HTTPFound
+from sqlalchemy.exc import DBAPIError
+from pyramid.authentication import AuthTktAuthenticationPolicy
+from pyramid.authorization import ACLAuthorizationPolicy
+import datetime
+HERE = os.path.dirname(os.path.abspath(__file__))
+
+DBSession = scoped_session(sessionmaker(extension=ZopeTransactionExtension()))
+Base = declarative_base()
+
+
+class Entry(Base):
+    __tablename__ = 'entries'
+    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
+    title = sa.Column(sa.Unicode(127), nullable=False)
+    text = sa.Column(sa.UnicodeText, nullable=False)
+    timestamp = sa.Column(
+        sa.DateTime, nullable=False, default=datetime.datetime.utcnow
+    )
+
+    @classmethod
+    def write(cls, title=None, text=None, session=None):
+        if session is None:
+            session = DBSession
+        instance = cls(title=title, text=text)
+        session.add(instance)
+        return instance
+
+    @classmethod
+    def all(cls, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).order_by(cls.timestamp.desc()).all()
+
+DATABASE_URL = os.environ.get(
+    'DATABASE_URL',
+    'postgresql://karenwong@localhost:5432/learning-journal'
+)
+
+
+def init_db():
+    engine = sa.create_engine(DATABASE_URL)
+    Base.metadata.create_all(engine)
+
+
+@view_config(route_name='home', renderer='templates/list.jinja2')
+def list_view(request):
+    entries = Entry.all()
+    return {'entries': entries}
+
+
+@view_config(route_name='add', renderer='templates/add.jinja2')
+def add_view(request):
+    return {}
+
+
+@view_config(route_name='detail', renderer='templates/detail.jinja2')
+def detail_view(request):
+    entries = Entry.all()
+    return {'entries': entries}
```

Comment:
Does your detail page show all the entries?  Why query for all of them.  It should only show one.